src/core/addonManager.js

```diff
@@ -24,9 +21,16 @@ export function getAddon(guid, { _mozAddonManager = window.navigator.mozAddonMan
     });
 }
 
-export function install(_url, eventCallback,
-  { _mozAddonManager = window.navigator.mozAddonManager } = {}) {
-  const url = addQueryParams(_url, { src: 'discovery-promo' });
+export function install(_url, eventCallback, {
+  _mozAddonManager = window.navigator.mozAddonManager,
+    src,
+} = {}) {
+  if (src === undefined) {
+    return new Promise(() => {
```

Comment:
I was cautious to do it that way because I wasn't 100% sure. It looks like the tests still pass with this so should be good.